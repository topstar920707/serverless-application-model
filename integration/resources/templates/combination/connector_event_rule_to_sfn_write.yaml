Resources:
  TriggerFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: nodejs16.x
      Handler: index.handler
      Timeout: 10  # in case eb has delay
      InlineCode: |
        const AWS = require('aws-sdk');
        
        exports.handler = async (event) => {
          const eb = new AWS.EventBridge();
          const response = await eb.putEvents({
            Entries: [{
              Source: process.env.EVENT_SOURCE,
              Detail: "{}",
              DetailType: "Test",
            }]
          }).promise();

          const sqs = new AWS.SQS();
          const data = await sqs.receiveMessage({
            QueueUrl: process.env.QUEUE_URL,
            WaitTimeSeconds: 5,
          }).promise();

          if (data.Messages.length == 0) {
            throw 'No messages in the queue!';
          }
        };
      Environment:
        Variables:
          QUEUE_URL: !Ref VerificationQueue
          EVENT_SOURCE: !Sub '${AWS::StackName}-test-event'
      Policies:
        - EventBridgePutEventsPolicy:
            EventBusName: default
        - SQSPollerPolicy:
            QueueName: !GetAtt VerificationQueue.QueueName

  EventRule:
    Type: AWS::Events::Rule
    Properties:
      Description: !Sub 'EventRule-${AWS::StackName}'
      EventPattern:
        source:
          - !Sub '${AWS::StackName}-test-event'
      Targets:
        - Arn: !GetAtt StateMachine.Arn
          Id: "Target"
          RoleArn: !GetAtt EBRole.Arn

  EBRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              Service: events.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::Serverless::StateMachine
    Properties:
      Type: EXPRESS
      Definition:
        StartAt: TryDoSomething
        States:
          TryDoSomething:
            Type: Task
            Resource: !Sub arn:${AWS::Partition}:states:::sqs:sendMessage
            Parameters:
              QueueUrl: !Ref VerificationQueue
              MessageBody: hello
            End: True
      Policies:
        - SQSSendMessagePolicy:
            QueueName: !GetAtt VerificationQueue.QueueName

  VerificationQueue:
    Type: AWS::SQS::Queue

  MyConnector:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: EventRule
      Destination:
        Id: StateMachine
      Permissions:
        - Write
