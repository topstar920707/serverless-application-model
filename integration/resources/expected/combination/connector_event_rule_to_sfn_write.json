[
  {
    "LogicalResourceId": "TriggerFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "TriggerFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "VerificationQueue",
    "ResourceType": "AWS::SQS::Queue"
  },
  {
    "LogicalResourceId": "MyConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  },
  {
    "LogicalResourceId": "EBRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "StateMachine",
    "ResourceType": "AWS::StepFunctions::StateMachine"
  },
  {
    "LogicalResourceId": "EventRule",
    "ResourceType": "AWS::Events::Rule"
  },
  {
    "LogicalResourceId": "StateMachineRole",
    "ResourceType": "AWS::IAM::Role"
  }
]